pkg/helm/helm_helpers.go

```diff
@@ -263,3 +271,53 @@ func AppendMyValues(valueFiles []string) ([]string, error) {
 	}
 	return valueFiles, nil
 }
+
+// CombineValueFilesToFile iterates through the input files and combines them into a single Values object and then
+// write it to the output file nested inside the chartName
+func CombineValueFilesToFile(outFile string, inputFiles []string, chartName string) error {
+	answer := chartutil.Values{}
+	for _, input := range inputFiles {
+		values, err := chartutil.ReadValuesFile(input)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to read helm values YAML file %s", input)
+		}
+		sourceMap := answer.AsMap()
+		CombineMapTrees(sourceMap, values.AsMap())
+		answer = chartutil.Values(sourceMap)
+	}
+	m := answer.AsMap()
+	answerMap := map[string]interface{}{
+		chartName: m,
+	}
+	answer = chartutil.Values(answerMap)
+	text, err := answer.YAML()
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal the combined values YAML files back to YAML")
+	}
+	err = ioutil.WriteFile(outFile, []byte(text), util.DefaultWritePermissions)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to save combined helm values YAML file %s", outFile)
+	}
+	return nil
+}
+
+// CombineMapTrees recursively copies all the values from the input map into the destination map preserving any missing entries in the destination
+func CombineMapTrees(destination map[string]interface{}, input map[string]interface{}) {
```

Comment:
I would move this into `pkg/utils/maps.go` that can be reused also in other parts of the oded.